source/tyr/tyr/helper.py

```diff
@@ -348,3 +350,21 @@ def hide_domain(email):
     if "@" not in email:
         return email
     return "{}******".format(email.split('@')[0])
+
+
+def repost_to_another_tyr(logger, url, content, instance_name):
+    content.seek(0)
+    file_to_post = {"file": (content.filename, content.stream, content.mimetype)}
+    url_secondary_tyr = '{}/jobs/{}'.format(url, instance_name)
+    nb_try = 0
+    while nb_try < MAX_NB_CALLS:
+        try:
+            resp = requests.post(url_secondary_tyr, files=file_to_post, verify=False)
+            logging.info('Info on posting data: {}'.format(resp.text))
+            if resp.status_code == 200:
+                return True
+            nb_try = nb_try + 1
+        except Exception as e:
+            logger.error("Error while posting data to secondary tyr: {i}: {e}".format(i=url_secondary_tyr, e=e))
+            return False
```

Comment:
Do we want to log such an error on each iteration of the loop? Maybe we want to catch the exception around the loop and inside the loop, only mention the number of the try, something like the following (not tested)?

```python
    try:
        while nb_try < MAX_NB_CALLS:
            resp = requests.post(url_secondary_tyr, files=file_to_post, verify=False)
            logging.info('Info on posting data: {}'.format(resp.text))
            if resp.status_code == 200:
                return True
            else:
                logger.warn("Failed to post data to secondary tyr [try {}]".format(nb_try)) 
            nb_try = nb_try + 1
    except Exception as e:
        logger.error("Error while posting data to secondary tyr: {i}: {e}".format(i=url_secondary_tyr, e=e))
        return False
```

Reply:
I just realized my proposition is not equivalent: if the request actually raise an exception (not an HTTP error), then we will never retry 🤔

Reply:
Maybe we need a two level exception catching.

```python
    try:
        while True:
            try:
                resp = requests.post(url_secondary_tyr, files=file_to_post, verify=False)
                logging.info('Info on posting data: {}'.format(resp.text))
                if resp.status_code == 200:
                    return True
            except Exception as e:
                if nb_try >= MAX_NB_CALLS:
                    raise e
            logger.warn("Failed to post data to secondary tyr [try {}]".format(nb_try))
            nb_try = nb_try + 1
    except Exception as e:
        logger.error("Error while posting data to secondary tyr: {i}: {e}".format(i=url_secondary_tyr, e=e))
        return False
```